### DISCUSSION TITLE
Discussion for Error/Warning Dialog in Quark-Jadx integration

### DISCUSSION THREAD

**Discussion by pulorsok** [quark-engine/quark-engine#174] [General]
## Discussion for Error/Warning Dialog in Quark-Jadx Integration
Hi @haeter525, there is a thing I want to discuss with you, 

In this [issue](https://github.com/quark-engine/gsoc2021-YuShiangDang/issues/2), I proposed a solution to solve one of the problems that [Jadx](https://github.com/skylot/jadx) asked me (the problem refers to [here](https://github.com/skylot/jadx/issues/1119#issuecomment-812013679))
> Therefore I strongly recommend to add some error dialog, as a button that just does nothing (except of failing silently) is bad user experience.

So I proposed to develop Error and Warning dialogs in [Quark-Jadx integration](https://github.com/skylot/jadx/pull/1135) to alert users that Quark does (or maybe) not work correctly. You can see the proposal detail [here](https://github.com/quark-engine/gsoc2021-YuShiangDang/issues/2#issuecomment-847640015).

However, there is a Warning scenario that I want to discuss.

- **Quark is not the latest version**.
The problem is that Quark currently doesn't have any option to provide the version directly. 
So I suggest developing a click option `quark --version` in [Quark-Engine](https://github.com/quark-engine/quark-engine) to show the Quark version. That helps developers and users more intuitive and easier to detect the version in programming or CLI.

If you agree with the solution, I will open an issue in [Quark-Engine](https://github.com/quark-engine/quark-engine) for adding `quark --version`.

**Comment by haeter525** [comment]
Hi @pulorsok 

I am pleased to review your writing. Here are the questions.

1. Is it supposed to be a GitHub discussion or an issue? You have used both words in the first two lines.
2. Which repo are you going to open? You have mentioned two issues about Quark-Engine and Jadx.
3. Can you explain the first scenarios you want to discuss further? The information isn't enough to start a discussion.
4. If the article mentions any issue or plan, provide a link to deliver the background. (Quark-Jadx integration and the "another issue")
5. If it is the first issue of the Error and Warning Dialog feature, I think you should give an illustration.

**Comment by pulorsok** [comment]
Hi @heater525, thank you for the review.

> 1. Is it supposed to be a GitHub discussion or an issue? You have used both words in the first two lines.
> 2. Which repo are you going to open? You have mentioned two issues about Quark-Engine and Jadx.

Sorry for the confusion, this discussion is not about the issue I want to open. 
It focuses on a problem and a solution that I provided and want to discuss with you first.

> 3. Can you explain the first scenarios you want to discuss further? The information isn't enough to start a discussion.

I think this scenario is quite complicated, so I will open another discussion to talk about that.

> 4. If the article mentions any issue or plan, provide a link to deliver the background. (Quark-Jadx integration and the "another issue")

I've added the link for each repo and issue.

> 5. If it is the first issue of the Error and Warning Dialog feature, I think you should give an illustration.

Since this discussion is focused on the solution, I only provide a brief of the feature. However, you can still see the detail [here](https://github.com/quark-engine/gsoc2021-YuShiangDang/issues/2#issuecomment-847640015) (including the illustration for the feature).

I have simplified the article above, please take a look.

**Comment by haeter525** [comment]
Hi @pulorsok, 

The idea of adding a flag for the version of Quark is awesome.

However, to show an informative prompt to users, three elements are needed.
1. The current version of Quark
2. The latest version of Quark
3. The actions that the user should take

Can you explain your plan for the rest two things in detail?
